packages/components/text/src/text.tsx

```diff
@@ -0,0 +1,248 @@
+import type { MessageDescriptor } from 'react-intl';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import requiredIf from 'react-required-if';
+import { useTheme } from '@emotion/react';
+import isNil from 'lodash/isNil';
+import { oneLine } from 'common-tags';
+import {
+  filterDataAttributes,
+  warnDeprecatedProp,
+} from '@commercetools-uikit/utils';
+import {
+  bodyStyles,
+  detailStyles,
+  headlineStyles,
+  subheadlineStyles,
+  wrapStyles,
+} from './text.styles';
+
+type TTextProps = {
+  intlMessage?: MessageDescriptor;
+  children?: React.ReactNode;
+};
+
+type TDeprecatedProps = {
+  // @deprecated
+  elementType: string;
+};
+
+const warnIfNoTextProps = (props: TTextProps, componentName) => {
+  if (!props.intlMessage && !React.Children.count(props.children)) {
+    console.error(
+      oneLine`
+        Warning: Failed prop type:
+        The prop \`intlMessage\` is marked as required in \`${componentName}\`,
+        but its value is \`undefined\`
+      `
+    );
+    console.error(
+      oneLine`
+        Warning: Failed prop type:
+        The prop \`children\` is marked as required in \`${componentName}\`,
+        but its value is \`undefined\`
+      `
+    );
+  }
+};
+
+const Text = (props: TTextProps) => (
+  <>
+    {props.intlMessage ? (
+      <FormattedMessage {...props.intlMessage} />
+    ) : (
+      props.children
+    )}
+  </>
+);
+Text.displayName = 'Text';
+
+type THeadlineProps = {
+  as?: 'h1' | 'h2' | 'h3';
+  // @deprecated
+  elementType?: 'h1' | 'h2' | 'h3';
+  title?: string | null;
+  truncate?: boolean;
+} & TTextProps;
+
+const Headline = (props: THeadlineProps) => {
+  const theme = useTheme();
+
+  if (props.elementType) {
+    warnDeprecatedProp(
+      'elementType',
+      'TextHeadline',
+      `\n \`elementType\` is deprecated. \n Please use "as" prop instead.`
+    );
+  }
+
+  warnIfNoTextProps(props, 'TextHeadline');
+
+  // For backwards compatibility
+  // we allow both `as` and `elementType` to be optional.
+  const HeadlineElement = props.as || props.elementType;
+
+  // however, if none of the prop is specified,
+  // we render plain text and set a warning on the log.
+  if (!HeadlineElement) {
+    console.error(
+      'ui-kit/Text: You attempt to render a TextHeadline without specifying `as` prop.'
+    );
+    return <Text intlMessage={props.intlMessage}>{props.children}</Text>;
+  }
```

Comment:
It makes sense to me to just render plain text, rather than falling back to e.g a `h1`

other suggestions welcome.
note, that at some point, we will remove `elementType` and make `as` required, so this shouldn't be needed.
my question is if we accept this as an intermediate solution.

Reply:
I think it's fine yes. Thanks!